code/unit_tests/power_tests.dm

```diff
@@ -71,22 +71,37 @@
 
 /datum/unit_test/area_power_tally_accuracy
 	name = "POWER: All areas must have accurate power use values."
+	var/list/channel_names = list("equip", "light", "environ")
+/obj/machinery/test_machine
+	power_channel = EQUIP
+	idle_power_usage = 1000
+	use_power = POWER_USE_IDLE
+
+/datum/unit_test/area_power_tally_accuracy/proc/check_power(var/area/A)
+	var/list/old_values = list(A.used_equip, A.used_light, A.used_environ)
+	A.retally_power()
+	var/list/new_values = list(A.used_equip, A.used_light, A.used_environ)
+	for(var/i in 1 to length(old_values))
+		if(abs(old_values[i] - new_values[i]) > 1) // Round because there can in fact be roundoff error here apparently.
```

Comment:
Not seeing a round here.

Reply:
The unrounded check would have been old_values == new_values; the way here accounts for roundoff error in math elsewhere.